File: fix_cards.py
import re

# Read both files
with open('vercel_index.html', 'r', encoding='utf-8') as f:
    vercel_html = f.read()
with open('index.html', 'r', encoding='utf-8') as f:
    local_html = f.read()

def find_matching_close_div(html, start_idx):
    """Find the matching closing </div> for the <div at start_idx"""
    depth = 0
    i = start_idx
    while i < len(html):
        if html[i:i+4] == '<div':
            depth += 1
        elif html[i:i+6] == '</div>':
            depth -= 1
            if depth == 0:
                return i + 6
        i += 1
    return -1

# --- 1. Extract grid from vercel ---
vercel_grid_start = vercel_html.find('<div class="grid">')
if vercel_grid_start == -1:
    # Try single quotes
    vercel_grid_start = vercel_html.find("<div class='grid'>")
    if vercel_grid_start == -1:
        print("ERROR: Could not find grid in vercel_index.html")
        # Look for card classes to debug
        card_idx = vercel_html.find('class="card')
        if card_idx > -1:
            print(f"Found card at index {card_idx}")
            print(f"Context: ...{vercel_html[max(0,card_idx-300):card_idx+50]}...")
        exit(1)

vercel_grid_end = find_matching_close_div(vercel_html, vercel_grid_start)
if vercel_grid_end == -1:
    print("ERROR: Could not find closing div for vercel grid")
    exit(1)

vercel_grid = vercel_html[vercel_grid_start:vercel_grid_end]
vercel_card_count = vercel_grid.count('class="card')
print(f"Extracted vercel grid: {len(vercel_grid)} chars, {vercel_card_count} cards")

# --- 2. Extract grid from local ---
local_grid_start = local_html.find('<div class="grid">')
if local_grid_start == -1:
    print("ERROR: Could not find grid in index.html")
    exit(1)

local_grid_end = find_matching_close_div(local_html, local_grid_start)
if local_grid_end == -1:
    print("ERROR: Could not find closing div for local grid")
    exit(1)

local_grid = local_html[local_grid_start:local_grid_end]
local_card_count = local_grid.count('class="card')
print(f"Local grid: {len(local_grid)} chars, {local_card_count} cards")

# --- 3. Extract vouch section from vercel ---
vercel_vouch_idx = vercel_html.find('class="vouch-grid"')
vercel_vouch_html = None
if vercel_vouch_idx > -1:
    vdiv_start = vercel_html.rfind('<div', 0, vercel_vouch_idx)
    vdiv_end = find_matching_close_div(vercel_html, vdiv_start)
    if vdiv_end > -1:
        vercel_vouch_html = vercel_html[vdiv_start:vdiv_end]
        print(f"Extracted vercel vouch grid: {len(vercel_vouch_html)} chars")

# --- 4. Add sold-out overlays ---
sold_out_overlay = '''<div class="card-pill">❌ Sold Out</div><div style="position:absolute;inset:0;background:rgba(0,0,0,0.6);z-index:10;display:flex;align-items:center;justify-content:center;pointer-events:none;"><span style="font-family:'Bebas Neue',sans-serif;font-size:1.1rem;font-weight:700;color:#e74c3c;letter-spacing:0.15em;border:2px solid #e74c3c;padding:0.4rem 1.2rem;">SOLD OUT</span></div>'''

def mark_sold_out(html, search_text):
    idx = html.find(search_text)
    if idx == -1:
        print(f"WARNING: Could not find '{search_text}'")
        return html
    
    # Find the card start
    card_start = html.rfind('<div class="card', 0, idx)
    if card_start == -1:
        print(f"WARNING: Could not find card div for '{search_text}'")
        return html
    
    # Find end of opening tag
    tag_end = html.find('>', card_start)
    card_tag = html[card_start:tag_end+1]
    
    # Update data-status
    if 'data-status=' in card_tag:
        new_card_tag = re.sub(r'data-status="[^"]*"', 'data-status="sold"', card_tag)
    else:
        new_card_tag = card_tag.replace('class="card', 'data-status="sold" class="card')
    
    # Check if already has sold overlay
    after_tag = html[tag_end+1:tag_end+201]
    if 'Sold Out' in after_tag or 'SOLD OUT' in after_tag:
        print(f"Already has sold overlay: '{search_text}'")
        return html
    
    html = html[:card_start] + new_card_tag + '\n      ' + sold_out_overlay + html[tag_end+1:]
    print(f"Marked SOLD OUT: '{search_text}'")
    return html

final_grid = vercel_grid

# Try with &amp; first (HTML entity), then with &
for search in ['Reaver 2.0, Primordium &amp; Kuronami', 'Reaver 2.0, Primordium & Kuronami']:
    new_grid = mark_sold_out(final_grid, search)
    if new_grid != final_grid:
        final_grid = new_grid
        break

for search in ['Kuronami, Narukami &amp; Champs 25', 'Kuronami, Narukami & Champs 25']:
    new_grid = mark_sold_out(final_grid, search)
    if new_grid != final_grid:
        final_grid = new_grid
        break

# --- 5. Replace grid in local HTML ---
new_html = local_html[:local_grid_start] + final_grid + local_html[local_grid_end:]

# --- 6. Replace vouch grid ---
if vercel_vouch_html:
    local_vouch_idx = new_html.find('class="vouch-grid"')
    if local_vouch_idx > -1:
        lv_start = new_html.rfind('<div', 0, local_vouch_idx)
        lv_end = find_matching_close_div(new_html, lv_start)
        if lv_end > -1:
            new_html = new_html[:lv_start] + vercel_vouch_html + new_html[lv_end:]
            print("Replaced vouch grid")

# --- 7. Write output ---
with open('index.html', 'w', encoding='utf-8') as f:
    f.write(new_html)

print(f"\n=== DONE ===")
print(f"New index.html: {len(new_html)} chars")

File: fix_cards.js
const fs = require('fs');

// Read both files
const vercelHTML = fs.readFileSync('vercel_index.html', 'utf8');
const localHTML = fs.readFileSync('index.html', 'utf8');

// --- 1. Extract the grid from vercel ---
// Find <div class="grid"> ... </div> that closes the grid (before </section>)
const vercelGridStart = vercelHTML.indexOf('<div class="grid">');
if (vercelGridStart === -1) {
  console.log('ERROR: Could not find <div class="grid"> in vercel_index.html');
  // Try alternate class names
  const alt1 = vercelHTML.indexOf('class="grid"');
  const alt2 = vercelHTML.indexOf("class='grid'");
  console.log('Alt search class="grid":', alt1);
  console.log("Alt search class='grid':", alt2);
  
  // Search for card patterns
  const cardIdx = vercelHTML.indexOf('class="card');
  console.log('First card class index:', cardIdx);
  if (cardIdx > -1) {
    console.log('Context around first card:', vercelHTML.substring(Math.max(0, cardIdx-200), cardIdx+100));
  }
  process.exit(1);
}

// Find the closing </div> for the grid - it's the one right before </section> or the "Choose Your Arsenal" style div
// We need to find the matching closing div. Let's find it by counting open/close divs
let depth = 0;
let vercelGridEnd = -1;
for (let i = vercelGridStart; i < vercelHTML.length; i++) {
  if (vercelHTML.substring(i, i+4) === '<div') {
    depth++;
  } else if (vercelHTML.substring(i, i+6) === '</div>') {
    depth--;
    if (depth === 0) {
      vercelGridEnd = i + 6;
      break;
    }
  }
}

if (vercelGridEnd === -1) {
  console.log('ERROR: Could not find closing </div> for grid');
  process.exit(1);
}

const vercelGrid = vercelHTML.substring(vercelGridStart, vercelGridEnd);
console.log('Extracted vercel grid, length:', vercelGrid.length);
console.log('Grid starts with:', vercelGrid.substring(0, 200));
console.log('Grid ends with:', vercelGrid.substring(vercelGrid.length - 200));

// Count cards in vercel grid
const vercelCardCount = (vercelGrid.match(/class="card/g) || []).length;
console.log('Number of cards in vercel grid:', vercelCardCount);

// --- 2. Extract the grid from local ---
const localGridStart = localHTML.indexOf('<div class="grid">');
if (localGridStart === -1) {
  console.log('ERROR: Could not find <div class="grid"> in index.html');
  process.exit(1);
}

let localDepth = 0;
let localGridEnd = -1;
for (let i = localGridStart; i < localHTML.length; i++) {
  if (localHTML.substring(i, i+4) === '<div') {
    localDepth++;
  } else if (localHTML.substring(i, i+6) === '</div>') {
    localDepth--;
    if (localDepth === 0) {
      localGridEnd = i + 6;
      break;
    }
  }
}

if (localGridEnd === -1) {
  console.log('ERROR: Could not find closing </div> for local grid');
  process.exit(1);
}

const localGrid = localHTML.substring(localGridStart, localGridEnd);
const localCardCount = (localGrid.match(/class="card/g) || []).length;
console.log('Number of cards in local grid:', localCardCount);

// --- 3. Now extract the vouch section from vercel ---
// Find the vouch-grid
const vercelVouchStart = vercelHTML.indexOf('class="vouch-grid"');
let vercelVouchSection = null;
let vercelVouchSectionHTML = null;

if (vercelVouchStart > -1) {
  // Back up to find the <div
  let vouchDivStart = vercelHTML.lastIndexOf('<div', vercelVouchStart);
  let vDepth = 0;
  let vouchDivEnd = -1;
  for (let i = vouchDivStart; i < vercelHTML.length; i++) {
    if (vercelHTML.substring(i, i+4) === '<div') {
      vDepth++;
    } else if (vercelHTML.substring(i, i+6) === '</div>') {
      vDepth--;
      if (vDepth === 0) {
        vouchDivEnd = i + 6;
        break;
      }
    }
  }
  if (vouchDivEnd > -1) {
    vercelVouchSectionHTML = vercelHTML.substring(vouchDivStart, vouchDivEnd);
    console.log('Extracted vercel vouch grid, length:', vercelVouchSectionHTML.length);
  }
}

// Find the local vouch-grid
const localVouchStart = localHTML.indexOf('class="vouch-grid"');
let localVouchSectionHTML = null;
let localVouchDivStart = -1;
let localVouchDivEnd = -1;

if (localVouchStart > -1) {
  localVouchDivStart = localHTML.lastIndexOf('<div', localVouchStart);
  let lvDepth = 0;
  for (let i = localVouchDivStart; i < localHTML.length; i++) {
    if (localHTML.substring(i, i+4) === '<div') {
      lvDepth++;
    } else if (localHTML.substring(i, i+6) === '</div>') {
      lvDepth--;
      if (lvDepth === 0) {
        localVouchDivEnd = i + 6;
        break;
      }
    }
  }
  if (localVouchDivEnd > -1) {
    localVouchSectionHTML = localHTML.substring(localVouchDivStart, localVouchDivEnd);
    console.log('Extracted local vouch grid, length:', localVouchSectionHTML.length);
  }
}

// --- 4. Now apply the sold-out overlays to the vercel grid for the specific accounts ---
// Mark these as sold out:
// 1. "Reaver 2.0, Primordium & Kuronami"
// 2. "Kuronami, Narukami & Champs 25"
let finalGrid = vercelGrid;

// Apply sold-out overlay to specific cards
const soldOutOverlay = `<div class="card-pill">❌ Sold Out</div><div style="position:absolute;inset:0;background:rgba(0,0,0,0.6);z-index:10;display:flex;align-items:center;justify-content:center;pointer-events:none;"><span style="font-family:'Bebas Neue',sans-serif;font-size:1.1rem;font-weight:700;color:#e74c3c;letter-spacing:0.15em;border:2px solid #e74c3c;padding:0.4rem 1.2rem;">SOLD OUT</span></div>`;

// Helper: Find a card containing specific text and add sold-out overlay
function markSoldOut(html, searchText) {
  const idx = html.indexOf(searchText);
  if (idx === -1) {
    console.log('WARNING: Could not find card with text:', searchText);
    return html;
  }
  
  // Find the card start (go backwards to find <div class="card)
  let cardStart = html.lastIndexOf('<div class="card', idx);
  if (cardStart === -1) {
    console.log('WARNING: Could not find card div start for:', searchText);
    return html;
  }
  
  // Update data-status to "sold"
  const cardTagEnd = html.indexOf('>', cardStart);
  const cardTag = html.substring(cardStart, cardTagEnd + 1);
  let newCardTag = cardTag;
  
  // Replace or add data-status="sold"
  if (newCardTag.includes('data-status=')) {
    newCardTag = newCardTag.replace(/data-status="[^"]*"/, 'data-status="sold"');
  } else {
    newCardTag = newCardTag.replace('class="card', 'data-status="sold" class="card');
  }
  
  // Insert sold-out overlay right after the opening card tag
  const afterCardTag = cardTagEnd + 1;
  const insertPoint = afterCardTag - cardStart;
  
  // Check if there's already a sold-out overlay
  const nextChunk = html.substring(afterCardTag, afterCardTag + 200);
  if (nextChunk.includes('Sold Out') || nextChunk.includes('SOLD OUT')) {
    console.log('Card already has sold-out overlay for:', searchText);
    return html;
  }
  
  html = html.substring(0, cardStart) + newCardTag + '\n      ' + soldOutOverlay + html.substring(afterCardTag);
  
  console.log('Marked as SOLD OUT:', searchText);
  return html;
}

finalGrid = markSoldOut(finalGrid, 'Reaver 2.0, Primordium &amp; Kuronami');
if (finalGrid === vercelGrid) {
  // Try without HTML entities
  finalGrid = markSoldOut(finalGrid, 'Reaver 2.0, Primordium & Kuronami');
}

const afterFirst = finalGrid;
finalGrid = markSoldOut(finalGrid, 'Kuronami, Narukami &amp; Champs 25');
if (finalGrid === afterFirst) {
  finalGrid = markSoldOut(finalGrid, 'Kuronami, Narukami & Champs 25');
}

// --- 5. Replace in local HTML ---
let newHTML = localHTML.substring(0, localGridStart) + finalGrid + localHTML.substring(localGridEnd);

// Replace vouch grid if both exist
if (vercelVouchSectionHTML && localVouchDivStart > -1 && localVouchDivEnd > -1) {
  // Recalculate positions since we changed the HTML
  const newLocalVouchStart = newHTML.indexOf('class="vouch-grid"');
  if (newLocalVouchStart > -1) {
    const newLocalVouchDivStart = newHTML.lastIndexOf('<div', newLocalVouchStart);
    let nvDepth = 0;
    let newLocalVouchDivEnd = -1;
    for (let i = newLocalVouchDivStart; i < newHTML.length; i++) {
      if (newHTML.substring(i, i+4) === '<div') {
        nvDepth++;
      } else if (newHTML.substring(i, i+6) === '</div>') {
        nvDepth--;
        if (nvDepth === 0) {
          newLocalVouchDivEnd = i + 6;
          break;
        }
      }
    }
    if (newLocalVouchDivEnd > -1) {
      newHTML = newHTML.substring(0, newLocalVouchDivStart) + vercelVouchSectionHTML + newHTML.substring(newLocalVouchDivEnd);
      console.log('Replaced vouch grid section');
    }
  }
}

// --- 6. Write output ---
fs.writeFileSync('index.html', newHTML, 'utf8');
console.log('\n=== DONE ===');
console.log('New index.html written successfully!');
console.log('New file size:', newHTML.length);
